includes/class-wc-amazon-payments-advanced-rest-api-controller.php

```diff
@@ -415,22 +416,20 @@ protected function authorize_order_v1( $order_id, $authorize_args = array() ) {
 	 *                                   WP_REST_Response instance.
 	 */
 	public function close_authorization( $request ) {
-		$order_post = $this->is_valid_order( $request['order_id'] );
-		if ( is_wp_error( $order_post ) ) {
-			return $order_post;
+		$or = $this->is_valid_order( $request['order_id'] );
+		if ( is_wp_error( $or ) ) {
+			return $or;
 		}
 
-		$order   = wc_get_order( $order_post->ID );
-		$version = WC_Amazon_Payments_Advanced::get_order_version( $order_post->ID );
+		$version = WC_Amazon_Payments_Advanced::get_order_version( $or->get_id() );
 		if ( 'v1' === strtolower( $version ) ) {
-			$error = $this->get_missing_authorization_id_request_error( $order_post );
+			$error = $this->get_missing_authorization_id_request_error( $or );
 			if ( is_wp_error( $error ) ) {
 				return $error;
 			}
 
-			$order_id = (int) $request['order_id'];
-			$auth_id  = get_post_meta( $order_id, 'amazon_authorization_id', true );
-			$resp     = WC_Amazon_Payments_Advanced_API_Legacy::close_authorization( $order_id, $auth_id );
+			$auth_id = $or->get_meta( 'amazon_authorization_id', true, true );
+			$resp    = WC_Amazon_Payments_Advanced_API_Legacy::close_authorization( $or->get_id(), $auth_id );
```

Comment:
Can we give this variable a better name, please?

Reply:
done!